WordPress/src/main/java/org/wordpress/android/ui/pages/PageParentFragment.kt

```diff
@@ -76,10 +86,63 @@ class PageParentFragment : Fragment() {
 
         saveButton = menu.findItem(R.id.save_parent)
         viewModel.isSaveButtonVisible.value?.let { saveButton?.isVisible = it }
+        searchAction = checkNotNull(menu.findItem(R.id.action_search)) {
+            "Menu does not contain mandatory search item"
+        }
+
+        initializeSearchView()
+    }
+
+    private fun initializeSearchView() {
+        searchAction.setOnActionExpandListener(object : OnActionExpandListener {
+            override fun onMenuItemActionExpand(item: MenuItem?): Boolean {
+                viewModel.onSearchExpanded(restorePreviousSearch)
+                return true
+            }
+
+            override fun onMenuItemActionCollapse(item: MenuItem?): Boolean {
+                viewModel.onSearchCollapsed()
+                return true
+            }
+        })
+
+        val searchView = searchAction.actionView as SearchView
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String): Boolean {
+                viewModel.onSearch(query)
+                return true
+            }
+
+            override fun onQueryTextChange(newText: String): Boolean {
+                if (restorePreviousSearch) {
+                    restorePreviousSearch = false
+                    searchView.setQuery(viewModel.lastSearchQuery, false)
+                } else {
+                    viewModel.onSearch(newText)
+                }
+                return true
+            }
+        })
+
+        val searchEditFrame = searchAction?.actionView.findViewById<LinearLayout>(R.id.search_edit_frame)
+        (searchEditFrame.layoutParams as LinearLayout.LayoutParams)
+                .apply { this.leftMargin = DisplayUtils.dpToPx(activity, -8) }
```

Comment:
We might want to consider moving the value into dimens.xml, wdyt?

SideNote: If you decide to move it to dimens.xml feel free to update even PagesFragment. Thanks!

Reply:
@malinajirka - didn't know the extent of my reach. I'll make this change.

Reply:
Actually I take that back, this is not a dimension per se, but an int interpretation. The DisplayUtils.dpToPx() takes an int value and uses the display metrics for calculation. I am leaving this one as is, but I am a bit rusty on converting dpToPx, so please advise.

Reply:
I think replacing `DisplayUtils.dpToPx(activity, -8)` with `resources.getDimensionPixelSize(R.dimens.xyz)` where `xyz = -8dp` would have the same result. All `dp` values take into account display metrics. Having said all that, let's not make this change now. You have a good point that it's not just a matter of replacing `-8` with `R.dimens.xyz`.